src/Microsoft.Azure.SignalR/ServerConnections/ServiceConnection.cs

```diff
@@ -434,6 +435,9 @@ private ClientConnectionContext RemoveClientConnection(string connectionId)
         {
             _connectionIds.TryRemove(connectionId, out _);
             _clientConnectionManager.TryRemoveClientConnection(connectionId, out var connection);
+#if NET7_0_OR_GREATER
```

Comment:
no need to have #if here since you already use #if when DI and here uses interface?

Reply:
I think this `#if` is necessary.

The behaviour wrapped by this `#if` is 
`_clientInvocationManager.Router.CleanupInvocationsByConnection(connectionId);` in `RemoveClientConnection`

When .NET < 7.0:
DI will pass in `DummyClientInvocationManager`. 
Look into this dummy manager,
```
public IRoutedClientResultsManager Router => throw new NotSupportedException();
```

`RemoveClientConnection` will be called in many other places. Without `#if`, the wrapped behaviour will throw exception.

Reply:
oh I c